controllers/storagecluster/storageconsumer.go

```diff
@@ -25,6 +25,13 @@ type storageConsumer struct{}
 var _ resourceManager = &storageConsumer{}
 
 func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (ctrl.Result, error) {
+	// TODO (leelavg): revisit in 4.20 whether this can be retired or not
+	// If the cluster was configured in provider mode pre4.19 a storageconsumer would already exist for serving storage to local services.
+	// We use allowRemoteStorageConsumers for finding the older provider mode as this field is made immutable from 4.19.
+	if storageCluster.Spec.AllowRemoteStorageConsumers {
```

Comment:
Don't we want to at least verify that a consumer exists in addition to the flag?

Reply:
What should the plan if one doesn't exist? Create internal storageconsumer?

Reply:
For now if it does not exists then return a reconcile error. There should not be any deployments of provider mode where there is no local consumer 

We might want to address creation of such consumer after doing multiple tests in the lab